core/data/reading.go

```diff
@@ -299,16 +290,18 @@ func readingByDeviceHandler(w http.ResponseWriter, r *http.Request) {
 		// Try to get device
 		// First check by name
 		var d models.Device
-		d, err := mdc.DeviceForName(deviceId)
+		d, err := mdc.CheckForDevice(deviceId)
```

Comment:
This check can be avoided using MetaDataCheck

Reply:
done